File: backend/firecracker-rootfs/languages/Dockerfile.go
FROM firecracker-base

# Install Go
RUN apk add --no-cache go

# Set up Go environment
ENV GOROOT=/usr/lib/go
ENV GOPATH=/go
ENV PATH=$GOPATH/bin:$GOROOT/bin:$PATH

# Create proper directories
RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"

# Test Go installation
RUN go version

File: backend/firecracker-rootfs/languages/Dockerfile.java
# languages/Dockerfile.java-fixed
FROM firecracker-base

# Install OpenJDK and its runtime dependencies
RUN apk add --no-cache \
        openjdk17 \
        openjdk17-jre

# Set up environment variables
ENV JAVA_HOME=/usr/lib/jvm/java-17-openjdk
ENV PATH="$JAVA_HOME/bin:$PATH"

RUN ln -sf $JAVA_HOME/lib/libjli.so /usr/lib/libjli.so

# Verify the installation works
RUN java -version && javac -version